packages/plugin-node-modules/sources/NodeModulesLinker.ts

```diff
@@ -0,0 +1,45 @@
+import {Installer, Linker, LinkOptions, MinimalLinkOptions}       from '@yarnpkg/core';
+import {FetchResult, Descriptor, Locator, Package, InstallStatus} from '@yarnpkg/core';
+import {PortablePath}                                             from '@yarnpkg/fslib';
+
+export class NodeModulesLinker implements Linker {
+  supportsPackage(pkg: Package, opts: MinimalLinkOptions): boolean {
+    return opts.project.configuration.get('nodeLinker') === 'node-modules';
+  }
+
+  async findPackageLocation(locator: Locator, opts: LinkOptions): Promise<PortablePath> {
+    return PortablePath.root;
```

Comment:
Note that `findPackageLocation` will probably be difficult to implement - it's meant to tell the `scriptUtils` where has been installed a specific package in a project, but due to the way the Node resolution works a package may need to be duplicated - so which one to use?

One way I could see it work (currently a bit hackish, but we can figure out later how to make it legit) would be to make `findPackageLocator` (the other function, defined below) return a unique locator for each installed package. So for example, even if `n_m/foo` and `n_m/bar/n_m/foo` have in theory the same locator, `findPackageLocator` would return something like:

```
locator-view:foo@1.0.0#/n_m/foo
locator-view:foo@1.0.0#/n_m/bar/n_m/foo
```

This way `findPackageLocation` would be able to figure out that the two locators point to different locations. The problem (and what makes it hackish) is that anything else using these locators wouldn't find them in the lockfile. They would need special support to know that when finding a `locator-view:` protocol, they need to unwrap it to get the "true" locator inside.